maven-universe/src/main/java/org/jboss/galleon/universe/maven/repo/MavenRepoManager.java

```diff
@@ -49,6 +49,8 @@ default Path resolve(String location) throws ProvisioningException {
 
     void resolve(MavenArtifact artifact) throws MavenUniverseException;
 
+    void resolveAll(List<MavenArtifact> artifacts) throws MavenUniverseException;
```

Comment:
Could you add a default implementation for resolveAll? It would iterate the list of artifacts and call the existing resolve method.